### PR TITLE
Updated the examples with better label and messages

### PR DESCRIPTION
**Pull request by ashishkcerner**
### Summary
Add contextual error message.
Provide contextual text to the legend to describe the functionality of fields.

**What was changed:**
Add contextual error message.
Provide contextual text to the legend to describe the functionality of fields.

**Why it was changed:**

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**
UXPLATFORM-10102
UXPLATFORM-10103
UXPLATFORM-10105 

Before Implemetation:- 
Screenshot 2024-01-08 at 4 27 18 PM

After Implementation:- 
Screenshot 2024-01-08 at 4 27 22 PM
5-9ce6-f389be0be516">

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/doc/date-input/example/MonthDayYearDateInputField.jsx [hunk 1]
```diff
@@ -9,16 +9,16 @@ const Example = () => {
...
       <DateInputField
-        legend="Legend text"
-        name="date-input-value"
+        legend="Patient's Birthdate"
+        name="patient-birthdate"
         value={value}
         onChange={(event, dateString) => setValue(dateString)}
         displayFormat="month-day-year"
-        error="Error message"
+        error="Please select a valid event date."
```
Comment (supreethmr) [inline packages/terra-framework-docs/src/terra-dev-site/doc/date-input/example/MonthDayYearDateInputField.jsx:17]:
```suggestion
        error="Please enter a valid event date."
```

Comment (ashishkcerner) [reply packages/terra-framework-docs/src/terra-dev-site/doc/date-input/example/MonthDayYearDateInputField.jsx]: Done https://github.com/cerner/terra-framework/pull/1961/commits/43cd7bdd8035ea750c596107643dbf3e63c648f2

File: packages/terra-framework-docs/src/terra-dev-site/doc/date-input/example/DayMonthYearDateInputField.jsx [hunk 2]
```diff
@@ -9,16 +9,16 @@ const Example = () => {
...
     <React.Fragment>
       <DateInputField
-        legend="Legend text"
+        legend="Event Date"
         name="date-input-value"
         value={value}
         onChange={(event, dateString) => setValue(dateString)}
         displayFormat="day-month-year"
-        error="Error message"
+        error="Please select a valid event date."
```
Comment (supreethmr) [inline packages/terra-framework-docs/src/terra-dev-site/doc/date-input/example/DayMonthYearDateInputField.jsx:17]: lets update help message in all the examples to have meaningful value like error message and legend text.

**Review by supreethmr — APPROVED** [review]

File: packages/terra-framework-docs/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,9 @@
 
 ## 1.54.0 - (January 2, 2024)
 
+* Changed
```
Comment (sugan2416) [inline packages/terra-framework-docs/CHANGELOG.md:7]: Move this under unreleased

Comment (ashishkcerner) [reply packages/terra-framework-docs/CHANGELOG.md:5]: done https://github.com/cerner/terra-framework/pull/1961/commits/664991f7f8e046ecae530655301af89601a6a024

**Review by sugan2416 — APPROVED** [review]

**Pull request merged**